### PR TITLE
refactor(search): refactor tests to use RTL instead of enzyme

### LINKED ISSUE #12153: Use RTL for testing in `Search/Search-test.js`
Please use [the script to scaffold out new test files](https://github.com/carbon-design-system/carbon/issues/10184#issuecomment-1282894448)

### PR DESCRIPTION
**Pull request by tay1orjones**
Closes #12153

#### Changelog

**Changed**

- Refactor `Search` and `ExpandableSearch` tests to use React Testing Library
- Fix a bug in Search - the `renderIcon` prop was not working

#### Testing / Reviewing

- Tests should pass

### CODE REVIEW

File: packages/react/src/components/Search/Search-test.js [hunk 1]
```diff
@@ -1,333 +1,157 @@
...
-        });
+    it('should respect labelText prop', () => {
+      render(<Search labelText="test-search" />);
 
-        it('is called', () => {
-          expect(onChange).toHaveBeenCalledWith(target.value);
-        });
-      });
+      expect(screen.getByRole('searchbox')).toBeInTheDocument();
+    });
```
Comment (francinelucca) [inline packages/react/src/components/Search/Search-test.js]: is testing that 'searchbox' is in the document sufficient to test for the labelText? 😅

**Review by francinelucca — CHANGES_REQUESTED** [review]

File: packages/react/src/components/Search/Search-test.js (hunk 1 above)
Comment (tay1orjones) [reply packages/react/src/components/Search/Search-test.js]: Zoinks! Great catch. I just updated it with a more thorough test

**Review by francinelucca — APPROVED** [review]
🔥

**Review by tw15egan — APPROVED** [review]
LGTM 👍 ✅

**Pull request merged**